polynote-server/src/main/scala/polynote/server/Server.scala

```diff
@@ -94,25 +93,21 @@ class Server(kernelFactory: Kernel.Factory.Service) extends polynote.app.App {
           |
           |""".stripMargin
 
-  override def main(args: List[String]): ZIO[Environment, Nothing, Int] = main.provideSomeLayer[Environment] {
-    Server.parseArgs(args).orDie andThen globalEnv.orDie
-  }
-
-  def main: ZIO[MainArgs with BaseEnv with ServerEnv, Nothing, Int] = for {
-    config       <- ZIO.access[Config](_.get)
-    _            <- Logging.info(s"Loaded configuration: $config")
-    wsKey         = config.security.websocketKey.getOrElse(UUID.randomUUID().toString)
-    _            <- Logging.warn(securityWarning)
-    _            <- Logging.info(banner)
-    _            <- serve(wsKey).orDie
-  } yield 0
 
+  def main: ZIO[AppEnv, String, Int] = {
+    for {
+      config       <- ZIO.access[Config](_.get[PolynoteConfig])
+      _            <- Logging.info(s"Loaded configuration: $config")
+      wsKey         = config.security.websocketKey.getOrElse(UUID.randomUUID().toString)
+      _            <- Logging.warn(securityWarning)
+      _            <- Logging.info(banner)
+      _            <- serve(wsKey).orDie
+    } yield 0
+  }.provideSomeLayer[AppEnv](IdentityProvider.layer.orDie)
 
-  private val globalEnv: ZLayer[BaseEnv with MainArgs, Throwable, ServerEnv] =
-    Server.loadConfig andThen (Interpreter.Factories.load ++ ZLayer.succeed(kernelFactory) ++ IdentityProvider.layer)
 
-  type ServerEnv = GlobalEnv with IdentityProvider
-  type RequestEnv = BaseEnv with ServerEnv with NotebookManager
+  type MainEnv = GlobalEnv with IdentityProvider with Has[NotebookRepository]
```

Comment:
Probably should have done the whole `Service`/type alias thing with `NotebookRepository` but I was too lazy to do it right now